skbio/diversity/beta/_unifrac.py

```diff
@@ -0,0 +1,170 @@
+
+    Parameters
+    ----------
+    u_counts, v_counts: list, np.array
+        Vectors of counts of OTUs for two samples. Must be equal length.
+    otu_ids: list, np.array
+        Vector of OTU ids corresponding to tip names in ``tree``. Must be the
+        same length as ``u_counts`` and ``v_counts``.
+    tree: skbio.TreeNode
+        Tree relating the OTUs in otu_ids. The set of tip names in the tree can
+        be a superset of ``otu_ids``, but not a subset.
+    normalized: boolean, optional
+        If ``True``, apply branch length normalization. Resulting distances
+        will then be in the range ``[0, 1]``.
+
+    Returns
+    -------
+    float
+        The weighted unifrac distance between the two samples.
+
+    Raises
+    ------
+    ValueError
+        If ``u_counts``, ``v_counts``, and ``otu_ids`` are not all equal in
+        length.
+    MissingNodeError
+        If an OTU id is provided that does not correspond to a tip in the
+        tree.
+
+    See Also
+    --------
+    unweighted_unifrac
+
+    Notes
+    -----
+    Unweighted unifrac was originally described in [1]_, which includes a
+    discussion of unweighted (qualitative) versus weighted (quantitiative)
+    diversity metrics.
+
+    References
+    ----------
+    .. [1] Lozupone, C. A., Hamady, M., Kelley, S. T. & Knight, R. Quantitative
+       and qualitative beta diversity measures lead to different insights into
+       factors that structure microbial communities. Appl. Environ. Microbiol.
+       73, 1576–1585 (2007).
+
+    """
+    u_observed_otus = {o: c for o, c in zip(otu_ids, u_counts) if c >= 1}
+    u_total_count = sum(u_counts)
+    v_observed_otus = {o: c for o, c in zip(otu_ids, v_counts) if c >= 1}
+    v_total_count = sum(v_counts)
+    if u_total_count == 0 and v_total_count == 0:
+        # boundary case where both communities have no members
+        return 0.0
+    u_observed_nodes = tree.observed_node_counts(u_observed_otus)
+    v_observed_nodes = tree.observed_node_counts(v_observed_otus)
+    weighted_unifrac = 0
+    D = 0
+    for o in set(u_observed_nodes) | set(v_observed_nodes):
+        # handle the case of o.length is None
+        b = o.length or 0
+        u_branch_weight = _sample_branch_weight(
+            u_observed_nodes[o], u_total_count)
+        v_branch_weight = _sample_branch_weight(
+            v_observed_nodes[o], v_total_count)
+        branch_weight = abs(u_branch_weight - v_branch_weight)
+        weighted_unifrac += b * branch_weight
+        if o.is_tip() and normalized:
+            d = o.accumulate_to_ancestor(tree.root())
+            normed_weight = u_branch_weight + v_branch_weight
+            D += (d * normed_weight)
+    if normalized:
+        return weighted_unifrac / D
+    else:
+        return weighted_unifrac
+
+
+def _sample_branch_weight(observed_nodes, total_count):
+    if total_count == 0:
+        return 0
```

Comment:
UniFrac is defined for that case, so it should be supported. If  ``u_counts`` is all zeros and ``v_counts`` is all zeros, the distance between those samples is ``0.0``. If ``u_counts`` is all zeros and ``v_counts`` has at least one non-zero value, unweighted unifrac and weighted unifrac with branch length normalization are 1.0, and weighted unifrac without branch length normalization will be >= 0.0 (depending on the branch lengths).